state-chain/chains/src/sol/sol_tx_core/address_derivation.rs

```diff
@@ -0,0 +1,149 @@
+//! Contains code for Deriving addresses and deposit channels for the Solana Environment.
+use cf_primitives::{chains::assets::sol::Asset as SolAsset, ChannelId};
+use sol_prim::AccountBump;
+
+use crate::{
+	deposit_channel::DepositChannel,
+	sol::{consts::*, AddressDerivationError, DerivedAddressBuilder, SolAddress},
+	Solana,
+};
+use core::str::FromStr;
+
+/// Derive address/deposit channels for a channel ID
+pub fn derive_deposit_channel(
+	channel_id: ChannelId,
+	_asset: SolAsset,
+	vault_program: SolAddress,
+) -> Result<DepositChannel<Solana>, AddressDerivationError> {
+	let seed = channel_id.to_le_bytes();
+	derive_address(seed, vault_program)
+		// .and_then(|(address, bump)| match asset {
+		// 	SolAsset::Sol => Ok((address, bump)),
+		// 	SolAsset::Usd => {
+		// 	    let mint_pubkey =
+		// 	Env::lookup_account(crate::sol::api::SolanaEnvAccountLookupKey::TokenMintPubkey)?;
+		// 	    derive_associated_token_account(address, mint_pubkey)
+		// 	},
+		// })
```

Comment:
Can this be removed?

Reply:
This is partial work for token support that has been implemented properly in the Token support PR.
This part has been cleaned up